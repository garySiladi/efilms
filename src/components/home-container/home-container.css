
:root {
  --home-container--margin-left: 50px;
  --home-container--padding: 20px;
  --slider-height: 350px;
  --home-container--transition-duration: 0.4s;
  --home-container--font-size: 2em;
  --expanded-offset: 430px;
}

.home {
  &-container {
    margin-left: var(--home-container--margin-left);
    padding: var(--home-container--padding) 0;
    transition-duration: var(--home-container--transition-duration);
  }

  & > div {
    margin: auto;
    width: 100%;
  }

  &-slider {
    font-family: EconSans;
    font-size: var(--home-container--font-size);
    margin-top: 0;
    padding: var(--home-container--padding);
    height: var(--slider-height);
    padding-bottom: 0;
    transition: transform var(--home-container--transition-duration);

    &--after-expanded {
      transform: translateY(var(--expanded-offset));
    }
  }
}
