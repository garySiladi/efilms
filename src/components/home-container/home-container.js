// @flow
import React from 'react';
import Slider from '../slider/slider';
import EpisodeSelected from '../episode-selected/episode-selected';
import './home-container.css';
import type { SeriesType } from '../app/app';

class HomeContainer extends React.Component {
  static isFullWidthSeries(seriesData) {
    return seriesData.thumbnail_size === 'full_width';
  }
  props: {
    series: Array<SeriesType>,
    selectedSeries: number,
    selectedEpisode: number,
    goToEpisode: boolean,
    closePopupFunction: Function,
    selectLowerSeries: Function,
  };
  render() {
    const {
      series,
      selectedSeries,
      selectedEpisode,
      goToEpisode,
      closePopupFunction,
      selectLowerSeries,
    } = this.props;
    if (!series[0]) return null;
    const selectedEpisodeData = series[selectedSeries].items[selectedEpisode];
    const episodeDetailsContainer = (
      <EpisodeSelected
        id={selectedEpisodeData.id}
        key={selectedEpisodeData.id}
        url={selectedEpisodeData.thumbnail.url}
        title={selectedEpisodeData.title}
        subtitle={selectedEpisodeData.subtitle}
        description={selectedEpisodeData.description}
        closePopupFunction={closePopupFunction}
        selectLowerSeries={selectLowerSeries}
        videoUrl={selectedEpisodeData.video_url}
        series={series}
        seriesId={selectedEpisodeData.series_id}
        isSelectedHomeContainer
        selectedSeries={selectedSeries}
        selectedEpisode={selectedEpisode}
        hideButtons={HomeContainer.isFullWidthSeries(series[selectedSeries])}
        isShown={goToEpisode}
        handleVideo={null}
      />
    );
    const homePageContent = series.map((data, index) =>
      (
        <Slider
          data={data.items}
          className="home-slider"
          sliderTitle={data.title}
          key={data.title}
          isSelected={index === selectedSeries}
          isAfterSelected={index > selectedSeries}
          selectedEpisode={selectedEpisode}
          isEpisodeExpanded={goToEpisode}
          isFullWidth={HomeContainer.isFullWidthSeries(data)}
        />
      ),
    );
    homePageContent.splice(selectedSeries + 1, 0, episodeDetailsContainer);
    const verticalOffset = -350;
    return (
      <div
        className="home-container"
        style={{ transform: `translate(0px, ${selectedSeries * verticalOffset}px)` }}
      >
        {homePageContent}
      </div>
    );
  }
}

export default HomeContainer;
