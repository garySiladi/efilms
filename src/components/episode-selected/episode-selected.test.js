// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, shallow } from 'enzyme';
import EpisodeSelected from './episode-selected';
import { mockData } from '../home-container/home-container.test';

jest.mock('./parts/episode-description', () =>
  jest.fn(() => <div>EpisodeDescription</div>),
);

jest.mock('../video-player-container/parts/video-player', () =>
  jest.fn(() => <div>VideoPlayer</div>),
);

function connectEvent(event, type, wrapper) {
  const changedEvent: Object = event;
  if (typeof type === 'number') {
    changedEvent.which = type;
  } else if (typeof type === 'string') {
    changedEvent.code = type;
  }
  const episodeSelected: Object = wrapper.instance();
  episodeSelected.handleKeyPress(event);
}

describe('HomeContainer ', () => {
  test('renders with data', () => {
    const tree : string = renderer.create(<EpisodeSelected
      id={3}
      url="https://cdn.twivel.io/uploads/economist/episode/thumbnail/141/episode_875X480.jpg"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={() => {}}
      selectedSeries={3}
      selectedEpisode={4}
      selectLowerSeries={() => {}}
      handleVideo={() => {}}
      series={mockData}
      videoUrl="https://cdn-films.economist.com/OCEANS/OCEANDEEP.m3u8"
      seriesId={7}
      isSelectedHomeContainer
      isShown
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('handles keyboard events when homeContainer is selected', () => {
    const episodeSelected = mount(<EpisodeSelected
      id={3}
      url="https://cdn.twivel.io/uploads/economist/episode/thumbnail/141/episode_875X480.jpg"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={() => {}}
      selectedSeries={3}
      selectedEpisode={4}
      selectLowerSeries={() => {}}
      handleVideo={null}
      series={mockData}
      videoUrl="https://cdn-films.economist.com/OCEANS/OCEANDEEP.m3u8"
      seriesId={7}
      isSelectedHomeContainer
      isShown
    />);
    const event = new Event('keyDown');
    // when home container is selected there are 2 buttons
    expect(episodeSelected.state().selectedItem).toEqual(0);
    connectEvent(event, 'Shift', episodeSelected);
    connectEvent(event, 'ArrowDown', episodeSelected);
    connectEvent(event, 'ArrowRight', episodeSelected);
    expect(episodeSelected.state().selectedItem).toEqual(1);
    connectEvent(event, 'ArrowRight', episodeSelected);
    expect(episodeSelected.state().selectedItem).toEqual(1);
    connectEvent(event, 'Enter', episodeSelected);
    connectEvent(event, 'ArrowLeft', episodeSelected);
    expect(episodeSelected.state().selectedItem).toEqual(0);
    connectEvent(event, 'ArrowLeft', episodeSelected);
    connectEvent(event, 'Enter', episodeSelected);
    connectEvent(event, 'ArrowUp', episodeSelected);
    connectEvent(event, 'Enter', episodeSelected);
    connectEvent(event, 'ArrowRight', episodeSelected);
    connectEvent(event, 'Backspace', episodeSelected);
    // when home container is unselected and there is just 1 button
    episodeSelected.setState({ isSelectedHomeContainer: false });
    expect(episodeSelected.state().selectedItem).toEqual(0);
    connectEvent(event, 'Enter', episodeSelected);
    connectEvent(event, 'ArrowLeft', episodeSelected);
    connectEvent(event, 'Enter', episodeSelected);
    connectEvent(event, 'ArrowRight', episodeSelected);
    connectEvent(event, 'Backspace', episodeSelected);
  });
  test('handles keyboard events when homeContainer is selected for WEBOS TV', () => {
    const episodeSelected = mount(<EpisodeSelected
      id={3}
      url="https://cdn.twivel.io/uploads/economist/episode/thumbnail/141/episode_875X480.jpg"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={() => {}}
      selectedSeries={3}
      selectedEpisode={4}
      selectLowerSeries={() => {}}
      handleVideo={() => {}}
      series={mockData}
      videoUrl="https://cdn-films.economist.com/OCEANS/OCEANDEEP.m3u8"
      seriesId={7}
      isSelectedHomeContainer
      isShown
    />);
    const event = new Event('keyDown');
    // when home container is selected there are 2 buttons
    connectEvent(event, 8, episodeSelected);
    connectEvent(event, 40, episodeSelected);
    connectEvent(event, 39, episodeSelected);
    connectEvent(event, 37, episodeSelected);
    connectEvent(event, 38, episodeSelected);
    connectEvent(event, 13, episodeSelected);
  });
  test('handles keyboard events when homeContainer is unselected', () => {
    const episodeSelected = mount(<EpisodeSelected
      id={3}
      url="https://cdn.twivel.io/uploads/economist/episode/thumbnail/141/episode_875X480.jpg"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={() => {}}
      selectedSeries={3}
      selectedEpisode={4}
      selectLowerSeries={() => {}}
      handleVideo={() => {}}
      series={mockData}
      videoUrl="https://cdn-films.economist.com/OCEANS/OCEANDEEP.m3u8"
      seriesId={7}
      isSelectedHomeContainer={false}
      isShown
    />);
    const event = new Event('keyDown');
    // when home container is unselected there is just 1 button
    expect(episodeSelected.state().selectedItem).toEqual(0);
    connectEvent(event, 'ArrowLeft', episodeSelected);
    connectEvent(event, 'ArrowUp', episodeSelected);
    connectEvent(event, 'ArrowRight', episodeSelected);
    connectEvent(event, 'ArrowDown', episodeSelected);
    connectEvent(event, 'Backspace', episodeSelected);
    connectEvent(event, 'Backspace', episodeSelected);
  });
  test('handles keyboard events without closePopupFunction', () => {
    const episodeSelected = mount(<EpisodeSelected
      id={3}
      url="https://cdn.twivel.io/uploads/economist/episode/thumbnail/141/episode_875X480.jpg"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      videoUrl="https://cdn-films.economist.com/OCEANS/OCEANDEEP.m3u8"
      seriesId={7}
      isSelectedHomeContainer
      closePopupFunction={() => {}}
      selectedSeries={3}
      selectedEpisode={4}
      selectLowerSeries={() => {}}
      handleVideo={() => {}}
      series={mockData}
      isShown
    />);
    const event = new Event('keyDown');
    // when home container is unselected and there is just 1 button
    expect(episodeSelected.state().selectedItem).toEqual(0);
    connectEvent(event, 'ArrowUp', episodeSelected);
    connectEvent(event, 'Backspace', episodeSelected);
  });
  test('mounts/unmounts', () => {
    const wrapper = shallow(<EpisodeSelected
      id={3}
      url="url xyz"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={() => {}}
      videoUrl=""
      seriesId={7}
      selectedSeries={3}
      selectedEpisode={4}
      selectLowerSeries={() => {}}
      handleVideo={() => {}}
      series={mockData}
      isSelectedHomeContainer
      isShown
    />);
    wrapper.unmount();
  });
  test('hides buttons', () => {
    const tree : string = renderer.create(<EpisodeSelected
      id={3}
      url="https://cdn.twivel.io/uploads/economist/episode/thumbnail/141/episode_875X480.jpg"
      title="title xyz"
      subtitle="subtitle xyz"
      description="description xyz"
      closePopupFunction={() => {}}
      videoUrl="https://cdn-films.economist.com/OCEANS/OCEANDEEP.m3u8"
      seriesId={7}
      selectedSeries={3}
      selectedEpisode={4}
      selectLowerSeries={() => {}}
      handleVideo={() => {}}
      series={mockData}
      isSelectedHomeContainer
      hideButtons={false}
      isShown
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
