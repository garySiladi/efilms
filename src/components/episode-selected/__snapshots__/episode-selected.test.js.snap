// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer  hides buttons 1`] = `
<div
  className="episode-selected episode-selected--expanded"
>
  <div
    className="episode-selected__teaser-wrapper"
  >
    <div
      className="video-container"
    >
      <div>
        VideoPlayer
      </div>
    </div>
    <div
      className="episode-selected__image episode-selected__image--selected"
    >
      <span
        aria-hidden={true}
        className="fa fa-chevron-left expand-button__arrow expand-button__arrow--first"
      />
      <span
        aria-hidden={true}
        className="fa fa-chevron-left expand-button__arrow expand-button__arrow--second"
      />
    </div>
  </div>
  <div
    className="episode-selected__info-container"
  >
    <div>
      EpisodeDescription
    </div>
    <div
      className="episode-buttons"
    >
      <div
        className="episode-buttons__learn"
      >
        Learn More
      </div>
    </div>
  </div>
</div>
`;

exports[`HomeContainer  renders with data 1`] = `
<div
  className="episode-selected episode-selected--expanded"
>
  <div
    className="episode-selected__teaser-wrapper"
  >
    <div
      className="video-container"
    >
      <div>
        VideoPlayer
      </div>
    </div>
    <div
      className="episode-selected__image episode-selected__image--selected"
    >
      <span
        aria-hidden={true}
        className="fa fa-chevron-left expand-button__arrow expand-button__arrow--first"
      />
      <span
        aria-hidden={true}
        className="fa fa-chevron-left expand-button__arrow expand-button__arrow--second"
      />
    </div>
  </div>
  <div
    className="episode-selected__info-container"
  >
    <div>
      EpisodeDescription
    </div>
  </div>
</div>
`;

exports[`HomeContainer:  renders correctly with data and goToEpisode prop enabled 1`] = `
<div
  className="home-container"
  style={
    Object {
      "transform": "translate(0px, -350px)",
    }
  }
>
  <div>
    Slider
  </div>
  <div>
    Slider
  </div>
  <div>
    EpisodeSelected
  </div>
  <div>
    Slider
  </div>
</div>
`;

exports[`HomeContainer:  renders correctly with data but without goToEpisode enabled 1`] = `
<div
  className="home-container"
  style={
    Object {
      "transform": "translate(0px, -350px)",
    }
  }
>
  <div>
    Slider
  </div>
  <div>
    Slider
  </div>
  <div>
    EpisodeSelected
  </div>
  <div>
    Slider
  </div>
</div>
`;

exports[`HomeContainer:  renders correctly without data 1`] = `null`;
